src/Mod/Show/TempoVis.py

```diff
@@ -93,33 +87,63 @@ def modifyVPProperty(self, doc_obj_or_list, prop_name, new_value):
         if App.GuiUp:
             if not hasattr(doc_obj_or_list, '__iter__'):
                 doc_obj_or_list = [doc_obj_or_list]
+            if not isinstance(prop_names,(list,tuple)):
+                prop_names = [prop_names]
             for doc_obj in doc_obj_or_list:
                 if not self.is3DObject(doc_obj):
                     continue
-                if not hasattr(doc_obj.ViewObject, prop_name):
-                    App.Console.PrintWarning("TempoVis: object {obj} has no attribute {attr}. Skipped.\n"
-                                             .format(obj= doc_obj.Name, attr= prop_name))
-                    continue # silently ignore if object doesn't have the property...
-
-                if doc_obj.Document is not self.document:  #ignore objects from other documents
-                    raise ValueError("Document object to be modified does not belong to document TempoVis was made for.")
-                oldval = getattr(doc_obj.ViewObject, prop_name)
-                setattr(doc_obj.ViewObject, prop_name, new_value)
-                self.restore_on_delete = True
-                if (doc_obj.Name,prop_name) not in self.data:
-                    self.data[(doc_obj.Name,prop_name)] = oldval
+                for prop_name in prop_names:
+                    if not hasattr(doc_obj.ViewObject, prop_name):
+                        App.Console.PrintWarning("TempoVis: object {obj} has no attribute {attr}. Skipped.\n"
+                                                .format(obj= doc_obj.Name, attr= prop_name))
+                        continue # silently ignore if object doesn't have the property...
+
+                    # Because the introduction of external objects, we shall now
```

Comment:
@realthunder is it actually used somewhere? I.e., temporarily changing a view property for an object from another project? Can you give an example of where it can come in handy?

Reply:
Yes, sure. Just create a body in one document, add some sketch, create some pad, pocket, whatever, save it. Then create another document, save it. And then create a link to the body in the first document. Finally, edit the sketch in the linked body. 

When editing from an external link, the `self.document` of a TempoVis is the active document, while the sketch object may live in other document, because of code [here](https://github.com/realthunder/FreeCAD/blob/dcf73d55e22dd0c96d62dab31bfe49faa7d95da2/src/Mod/Sketcher/Gui/ViewProviderSketch.cpp#L5662). And the reason to create TempoVis with the active document rather than the owner document of the sketch is so that it can correctly hide/show dependency in the active document, where the editing objects are actually shown.

Reply:
OK, thanks. I'm going to experiment with it a bit, to understand it better. FYI, [i'm reworking TempoVis](https://forum.freecadweb.org/viewtopic.php?f=10&t=38592), and I have to incorporate that. And thus I need a thorough understanding of use cases.

Reply:
I'll write a summary of my changes in that thread shortly.